stdlib/SparseArrays/src/linalg.jl

```diff
@@ -1028,9 +1028,16 @@ function opnorm(A::SparseMatrixCSC, p::Real=2)
     m, n = size(A)
     if m == 0 || n == 0 || isempty(A)
         return float(real(zero(eltype(A))))
-    elseif m == 1 || n == 1
-        # TODO: compute more efficiently using A.nzval directly
-        return opnorm(Array(A), p)
+    elseif m == 1
+        if p == 1
+            return norm(A.nzval, Inf)
+        elseif p == 2
+            return norm(A.nzval, 2)
+        elseif p == Inf
+            return norm(A.nzval, 1)
+        end
```

Comment:
There's an `else`-branch missing, isn't it? What if `m == 1` but `p` is neither 1, 2, nor `Inf`? I guess it should be either `norm(A.nzval, p/(p-1))`, or `opnorm(A.nzval', p)`.

Reply:
Oh, that's correct. I assumed that `opnorm` does only support `p = 1, 2, Inf` as the last line of the method suggests:
```
throw(ArgumentError("invalid operator p-norm p=$p. Valid: 1, Inf"))
```
But it used to work for arbitrary values of `p` in the case where `A` is a sparse `m x 1` or `1 x n` matrix, so it should probably be kept this way. I'll see what I can do.

Reply:
Yes, but that's the case when neither `m == 1` nor `n == 1`, i.e., for actual matrices. If any of the two is 1, then `opnorm(Array(A), p)` is computed, without further checks on `p`. I guess you could have
```julia
elseif m == 1
    norm(A.nzval, p / (p - 1))
elseif n == 1
    norm(A.nzval, p)
else
     # the matrix case ...
```

Reply:
Actually, my previous statement was wrong: The current implementation of `opnorm` does not support any values of `p` other than `1, 2, Inf`, not even in the case where `A` is a `m x 1` or `1 x n` sparse matrix:
```julia
using LinearAlgebra
using SparseArrays
A = SparseMatrixCSC([1.0 2.0 0.0 0.3])
opnorm(A, 3)
```
gives
```
ERROR: ArgumentError: invalid p-norm p=3. Valid: 1, 2, Inf
Stacktrace:
 [1] opnorm at /build/julia/src/julia-1.1.1/usr/share/julia/stdlib/v1.1/LinearAlgebra/src/generic.jl:583 [inlined]
 [2] opnorm(::SparseMatrixCSC{Float64,Int64}, ::Int64) at /build/julia/src/julia-1.1.1/usr/share/julia/stdlib/v1.1/SparseArrays/src/linalg.jl:910
 [3] top-level scope at none:0
```

So the question is: Keep it that way (only allowing `p = 1, 2, Inf`) or allowing more general `p` values for the case where `A` is a sparse `m x 1` or `1 x n` matrix?

@dkarrasch What would be the restrictions on `p` with your suggestion `norm(A.nzval, p / (p - 1))`? Simply allowing `p > 0` with special cases for `p = 1` and `p = Inf`?

Reply:
Ok, I confused myself. My apologies. The reason is that `opnorm(A::AbstractArray{T,2} where T, p::Real)` is not implemented for any other `p` than `1, 2, Inf`, even if `m == 1 || n == 1`. In the latter case, the opnorm reduces to a (dual) vector norm, which technically is defined for other `p`s. I guess that, for consistency, we should leave it at the three cases. However, you will need to check for `p in (1, 2, Inf)` in the line right below this discussion.